### PR TITLE
feat(product): add discount field to DaffProduct

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
DaffProduct has no discount field.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/product/testing/src/factories/composite-product.factory.ts [hunk 1]
```diff
@@ -7,10 +7,16 @@ import { DaffModelFactory } from '@daffodil/core/testing';
...
  */
 export class MockCompositeProduct implements DaffCompositeProduct {
+	private stubPrice = faker.random.number(1500);
+	private stubDiscount = faker.random.number({min: 0, max: this.stubPrice - 1});
 	type = DaffProductTypeEnum.Composite;
 	id = faker.random.number(10000).toString();
 	url = faker.random.alphaNumeric(16);
-  price = faker.random.number(1500).toString();
+	price = this.stubPrice.toString();
+	discount = {
```
Comment (griest024) [inline libs/product/testing/src/factories/composite-product.factory.ts:16]: needs test

File: libs/product/src/models/configurable-product.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export interface DaffConfigurableProductVariant {
 	appliedAttributes: DaffProductVariantAttributesDictionary;
 	id: string;
 	price: string;
+	discount: DaffProductDiscount;
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/src/models/configurable-product.ts:21]: It's for the configurable product variants. I had no rational for making it required. But looking at it now, I guess making it required would make it so the app dev wouldn't need to null check the discount field.

[the remark this answers is not in the document]
Comment (griest024) [reply libs/product/src/models/configurable-product.ts:21]: I figured it was something like that. I was just wondering why the base product had a nullable `discount` and this one was required.

[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/src/models/configurable-product.ts:21]: Oh, I think a lot of the base product nullable fields shouldn't really be nullable. I was just following suit there basically

File: libs/product/testing/src/factories/composite-product.factory.ts (hunk 1 above)
Comment (lderrickable) [reply libs/product/testing/src/factories/composite-product.factory.ts:16]: added

**Pull request merged**